pkg/agent/manager/cache/cache.go

```diff
@@ -355,11 +360,39 @@ func (c *Cache) UpdateSVIDs(update *UpdateSVIDs) {
 			telemetry.SPIFFEID: record.entry.SpiffeId,
 		})
 		log.Debug("SVID updated")
+
+		// Registration entry is updated, remove it from stale map
+		delete(c.staleEntries, entryID)
 	}
 
 	c.notifyBySelectors(notifySet)
 }
 
+// GetStaleEntries obtains a list of stale entries
+func (c *Cache) GetStaleEntries() map[*common.RegistrationEntry]time.Time {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	staleEntries := make(map[*common.RegistrationEntry]time.Time)
+	for entryID := range c.staleEntries {
+		cachedEntry, ok := c.records[entryID]
+		if !ok {
+			c.log.WithField(telemetry.RegistrationID, entryID).Warn("stale entry not found")
```

Comment:
Log messages should be sentence formatted

What does this warning message mean? Under what conditions is it hit? If I'm an operator and I see this message, what should I do?

Reply:
It is happening in case we have a `stale` entry in our map, but it is not longer in our cache, because it was removed

Reply:
If this exists at all, it should be purely defensive. The path that deletes a record removes the stale entry. I wonder if this shouldn't just log that there was a problem and in addition "Please file a bug" and not bother messing with locks to try and clean up the map.

Reply:
yes, it is defensive to have dead entries in our staleEntries, so you suggest:
- keep RLock
- update error message to fill ask users to fill a bug, something like: `Could not found stale  registration entry. Please file a bug`

Reply:
Is it worth logging at all?

Reply:
personally I prefer to logging, It must not happens, but in case it happens it is good we know something is wrong

Reply:
It is good for a developer, to know something is wrong, but is it good for an operator who isn't affected by it and can't really do anything about it?

Reply:
I don't really care if we log it or not. I doubt we'll ever see it so I'd personally be fine if we didn't. That being said, logging is an appropriate channel to let a developer know that something is wrong for something that does not impact, nor can be reconciled by, the operator. But in these instances it's probably better to be done at DEBUG level, instead of WARN.

Reply:
right now it is an`error`, if both of you are  ok, I can move it to debug, what do you think about that?

Reply:
I think debug is a good compromise

Reply:
updated!